pages/docs/concepts/asyncapi document/variable-url.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Variable URL
+weight: 275
+---
+
+[Event-driven architecture](/docs/tutorials/getting-started/event-driven-architectures) benefits from a defined base URL and rules for URL variables. You can easily manage multiple API endpoints, handling various server configurations and environments.
+
+URL variables are placeholders for values you can replace during runtime. AsyncAPI enables you to construct dynamic URLs with query parameters and additional information while enhancing the flexibility and maintainability of your API specifications.
+
+## Add dynamic variables
+
+To add variables to the URL using AsyncAPI between two servers. You can use the `server.url` and `components.serverVariables` fields to enable reusable variables across multiple servers.
+
+The diagram below describes how to use variable urls in AsyncAPI.
+
+```mermaid
+graph TD
+
+    A[AsyncAPI Server]
+    B[Components]
+    C[Server URL]
+
+style A fill:#47BCEE,stroke:#47BCEE;
+style B fill:#47BCEE,stroke:#47BCEE;
+style C fill:#47BCEE,stroke:#47BCEE;
+
+  B -->|Defines serverVariables| C
+  C -->|Reusable variables| A
+```
+
+### Servers section
+
+Define the servers section in your AsyncAPI document, and include the base URLs for your API servers. Use placeholders enclosed in curly braces {} to represent the variables in the server URL. For example:
+
+```yaml
+servers:
+  production:
+    url: 'https://{subdomain}.example.com:{port}/v1'
+    description:
+      Production server:
+      variables:
+        domain:
+          enum:
+            - development
+            - staging
+            - production
+        port:
+          default: '8080'
+```
+
+### `serverVariables` section
+
+Define the components.serverVariables section in your AsyncAPI document. For each variable used in the server URLs, provide a default value and an optional description:
+
+```yaml
+components:
+  serverVariables:
+    domain:
+      default: 'api'
+      description: The domain of the API server port
+        default: '8080'
+        description: The port of the API server
+```
+
+### Define domain and port variables
```

Comment:
This is very specific and it might make people think that server variables are for defining subdomains and ports. You can but this is just one of the multiple usages.